connection.go

```diff
@@ -16,9 +16,11 @@ type Connection struct {
 	session          *Session
 	localClientID    string
 	remoteClientID   string
-	windowSize       uint32
 	sendWindowUpdate chan struct{}
 
+	winSizeMu  sync.RWMutex
```

Comment:
Don't add this. The lock is not only used to protect single variable (atomic can handle that simple case), but also used to make sure nothing is changed during a block of code logic.

Reply:
Yes, modified.